crates/tiny/src/main.rs

```diff
@@ -142,10 +144,26 @@ fn run(
         for server in servers.iter().cloned() {
             tui.new_server_tab(&server.addr, server.alias);
 
+            let tls = server.tls;
+            let sasl_auth = if let Some(sasl_auth) = &server.sasl_auth {
```

Comment:
How can I simplify this? 🤔

Reply:
Better or worse?
```
let sasl_auth = server
    .sasl_auth
    .as_ref()
    .map(|sasl| {
        sasl_from_config(tls, sasl)
            .map_err(|e| {
                tui.add_client_err_msg(
                    &format!("SASL error for server [{}]: {}", server.addr, e),
                    &MsgTarget::Server { serv: "mentions" },
                );
            })
            .ok()
    })
    .flatten();
```

Reply:
To me current version doesn't look too bad. I don't like the `flatten()` in the new version, it becomes difficult to read as there are multiple `flatten` methods (for `Option`, `Result`, `Iterator`), and you need to figure out the types to understand the code.

Another alternative:

```rust
let sasl_auth =
    server
        .sasl_auth
        .map(|sasl_auth| match sasl_from_config(tls, &sasl_auth) {
            Ok(sasl) => Some(sasl),
            Err(e) => {
                tui.add_client_err_msg(
                    &format!("SASL error for server [{}]: {}", server.addr, e),
                    &MsgTarget::Server { serv: "mentions" },
                );
                None
            }
        });
```

Reply:
Because we want to report the error in the server tab I'm going to leave the original since it avoids the partial borrow, which can only be prevented by moving the initialization of `sasl_auth` above `tui.new_server_tab()` (takes ownership of `server.alias`).